### ISSUE TITLE
VMHostVDSwitchMigration DSC Resource fails to migrate physical network adapters in some circumstances

### ISSUE THREAD

**Issue by mikedonleyuoy** [vmware/dscr-for-vmware#281]
On an ESXi host with 2 or more physical network adapters and no disconnected physical network adapters, the VMHostVDSwitchMigration DSC resource fails to migrate the last connected physical adapter to the VDS

On a ESXi host with a single physical adapter, the migration is successful.

If the host has one or more disconnected physical interfaces, the migration is successful.

Using the following configuration (vmnic0 & 1 are connected)
```
    VMHostVDSwitchMigration VMHostVDSwitchMigrationmikey {
        Credential = $Credential 
        PhysicalNicNames = @('vmnic0','vmnic1')
        Server = $AllNodes.Server 
        VdsName = 'MyVDSwitch'
        VMHostName = $AllNodes.VMHostName
        VMKernelNicNames = 'vmk0'
        PortGroupNames = 'Management Network'
```

The error produced is 
```
VERBOSE: [MYHOST]: LCM:  [ Start  Set      ]  [[VMHostVDSwitchMigration]VMHostVDSwitchMigration]
VERBOSE: [MYHOST]:                            [[VMHostVDSwitchMigration]VMHostVDSwitchMigration] 06/10/2020 11:19:40 Get-VMHost Finished execution
VERBOSE: [MYHOST]:                            [[VMHostVDSwitchMigration]VMHostVDSwitchMigration] 06/10/2020 11:19:40 Get-VMHostNetworkAdapter Finished execution
VERBOSE: [MYHOST]:                            [[VMHostVDSwitchMigration]VMHostVDSwitchMigration] 06/10/2020 11:19:40 Get-VMHostNetworkAdapter Finished execution
VERBOSE: [MYHOST]:                            [[VMHostVDSwitchMigration]VMHostVDSwitchMigration] 06/10/2020 11:19:40 Get-VMHostNetworkAdapter Finished execution
VERBOSE: [MYHOST]:                            [[VMHostVDSwitchMigration]VMHostVDSwitchMigration] 06/10/2020 11:19:40 Get-VDPortgroup Finished execution
VERBOSE: [MYHOST]:                            [[VMHostVDSwitchMigration]VMHostVDSwitchMigration] 06/10/2020 11:19:40 Add-VDSwitchPhysicalNetworkAdapter Finished execution
VERBOSE: [MYHOST]: LCM:  [ End    Set      ]  [[VMHostVDSwitchMigration]VMHostVDSwitchMigration]  in 1.4550 seconds.
PowerShell DSC resource VMware.vSphereDSC  failed to execute Set functionality with error message: Could not migrate Physical Network Adapters vmnic1  and VMKernel Network Adapters vmk0 to Distributed Switch
MyVDSwitch. For more information: Cannot validate argument on parameter 'VMHostPhysicalNic'. The argument is null, empty, or an element of the argument collection contains a null value. Supply a collection
that does not contain any null values and then try the command again.
    + CategoryInfo          : InvalidOperation: (:) [], CimException
    + FullyQualifiedErrorId : ProviderOperationExecutionFailure
    + PSComputerName        : localhost

```
The problem seems to be in VMware.vSphereDSC.psm1 GetConnectedPhysicalNetworkAdapters (line 14562).  If the host has no disconnected adapters, it seems to return an blank array.  This means when connected adapters and unconnected adapters are concatented on line 14794 and 14842, the final entry is seen as being 'vmnic1 ' (note the extra space) rather than 'vmnic1'.

Changing the return value for this function to $null doesn't fix the problem.

This could possibly be fixed by doing this
```
    $physicalNetworkAdaptersToMigrate = $connectedPhysicalNetworkAdapters
    if ($disconnectedPhysicalNetworkAdapters.count -gt 0) {
        $physicalNetworkAdaptersToMigrate += $disconnectedPhysicalNetworkAdapters
    }
```
to replace 
```
    $physicalNetworkAdaptersToMigrate = $connectedPhysicalNetworkAdapters + $disconnectedPhysicalNetworkAdapters
```

Not sure if there is a more elegant way?

Testing Environment
vCenter 7.0.0 build 16749670
ESXi 6.7.0 build 16713306
vSphere DSC Module v2.1.0.71

**Comment by jatinpurohit92** [comment]
I have also experienced the same. IMO issue is with  Add-VDSwitchPhysicalNetworkAdapter to get the work done.
If i pass both the nics together, it fails. But if i pass one NIC
![image (7)](https://user-images.githubusercontent.com/55919052/95216682-fdc4d900-080f-11eb-8fcc-9519720eaf2f.png)
 at a time it succeed.

**Comment by mikedonleyuoy** [comment]
That was my first workaround, but the DSC engine then complained because I had two configuration blocks for the same resource.  That got me digging!  My feeling is this could be an upstream bug in PowerCLI as if you do the same concatentation on a String, it *seems* to work, but this is 'VMware.VimAutomation.ViCore.Impl.V1.Host.Networking.Nic.PhysicalNicImpl' type.

**Comment by mikedonleyuoy** [comment]
It looks like possibly changing

`$physicalNetworkAdaptersToMigrate = $connectedPhysicalNetworkAdapters + $disconnectedPhysicalNetworkAdapters`

to

`[array]$physicalNetworkAdaptersToMigrate = $connectedPhysicalNetworkAdapters + $disconnectedPhysicalNetworkAdapters`

May also work, but I've not tested that as much.

**Comment by SimeonGerginov** [comment]
@mikedonleyuoy could you share these additional details about your environment:

1. Is **vmnic1** Physical Network Adapter part of Standard Switch or only **vmnic0** is part of a Standard Switch ?
2. If both nics are part of a Standard Switch, are they active, standby or unused ?

**Comment by mikedonleyuoy** [comment]
@SimeonGerginov In my test setup both vmnic0 and vmnic1 were part of the Standard Switch, and both nics are set as active.

**Comment by SimeonGerginov** [comment]
The bug was fixed with PR #286, so the migration should work as expected with the latest version of the module.
